app/src/main/java/org/listenbrainz/android/ui/screens/listens/ListensScreen.kt

```diff
@@ -60,24 +84,50 @@ fun ListensScreen(
         },
         playListen = {
             socialViewModel.playListen(it)
-        }
+        },
+        snackbarState = snackbarState
     )
 }
 
+private enum class ListenDialogBundleKeys {
+    PAGE,
+    EVENT_INDEX;
+    companion object {
+        fun listenDialogBundle(page: Int, eventIndex: Int): Bundle {
+            return Bundle().apply {
+                putInt(PAGE.name, page)
+                putInt(EVENT_INDEX.name, eventIndex)
+            }
+        }
+    }
+}
+
+
 
 @OptIn(ExperimentalFoundationApi::class)
 @Composable
 fun ListensScreen(
     scrollRequestState: Boolean,
     onScrollToTop: (suspend () -> Unit) -> Unit,
+    socialViewModel: SocialViewModel = hiltViewModel(),
```

Comment:
Please lift view-models out of this layer to the upper layer. This makes previews possible. Pass all the functions and states you need as parameters.